source/type/pb_converter.cpp

```diff
@@ -623,6 +623,32 @@ void PbCreator::Filler::fill_pb_object(const nt::Line* l, pbnavitia::Line* line)
         pb_property->set_value(property.second);
     }
 
+    if (dump_message == DumpMessage::Yes && dump_line_section_message) {
+        /*
+         * Here we dump the impacts which impact LineSection.
+         * We could have link the LineSection impact with the line, but that would change the code and
+         * the behavior too much.
+         * */
+        std::set<boost::shared_ptr<nt::disruption::Impact>> added_impact;
+        auto pred = [&](const nt::VehicleJourney& vj) {
+            for(const auto& impact: vj.meta_vj->impacted_by) {
+                auto impact_ptr = impact.lock();
+                if (! impact_ptr)
+                    continue;
+                for (const auto& entity: impact_ptr->informed_entities()) {
+                    if (boost::get<nt::disruption::LineSection>(&entity) &&
+                            added_impact.insert(impact_ptr).second ) {
+                        fill_messages(vj.meta_vj, line);
+                        return true;
```

Comment:
we can have multiple line_sections on the impact, but we add only once the impact :)